projects/igniteui-angular/src/lib/grids/grid-navigation.service.ts

```diff
@@ -148,45 +151,53 @@ export class IgxGridNavigationService {
 
         this.performHeaderKeyCombination(this.currentActiveColumn, key, shift, ctrl, alt, event);
         if (shift || alt || (ctrl && (key.includes('down') || key.includes('down')))) { return; }
-        !this.grid.hasColumnGroups ? this.horizontalNav(event, key, -1) : this.handleMCHeaderNav(key, ctrl);
+        !this.grid.hasColumnGroups ? this.horizontalNav(event, key, -1, 'headerCell') : this.handleMCHeaderNav(key, ctrl);
     }
 
-    protected horizontalNav(event: KeyboardEvent, key: string, rowIndex: number) {
+    protected horizontalNav(event: KeyboardEvent, key: string, rowIndex: number, tag: GridKeydownTargetType) {
         const ctrl = event.ctrlKey;
         if (!HORIZONTAL_NAV_KEYS.has(event.key.toLowerCase())) { return; }
         event.preventDefault();
         this.activeNode.row = rowIndex;
         if (rowIndex > 0) {
-            if (this.emitKeyDown(GridKeydownTargetType.summaryCell, this.activeNode.row, event)) {
+            if (this.emitKeyDown('summaryCell', this.activeNode.row, event)) {
                 return;
             }
         }
+        let newActiveNode = this.activeNode.column;
         if ((key.includes('left') || key === 'home') && this.activeNode.column > 0) {
-            this.activeNode.column = ctrl || key === 'home' ? 0 : this.activeNode.column - 1;
+            newActiveNode = ctrl || key === 'home' ? 0 : this.activeNode.column - 1;
         }
         if ((key.includes('right') || key === 'end') && this.activeNode.column < this.lastColumnIndex) {
-            this.activeNode.column = ctrl || key === 'end' ? this.lastColumnIndex : this.activeNode.column + 1;
+            newActiveNode = ctrl || key === 'end' ? this.lastColumnIndex : this.activeNode.column + 1;
         }
+
+        this.setActiveNode({row: this.activeNode.row, column: newActiveNode}, tag);
         this.performHorizontalScrollToCell(this.activeNode.column);
     }
 
     focusTbody(event) {
         const gridRows = this.grid.verticalScrollContainer.totalItemCount ?? this.grid.dataView.length;
         if (gridRows < 1) { this.activeNode = null; return; }
         if (!this.activeNode || this.activeNode.row < 0 || this.activeNode.row > gridRows - 1) {
-            this.activeNode = { row: 0, column: 0 };
+            const item = this.grid.dataView[0];
+            const type = this.grid.isGroupByRecord(item) ? 'groupRow' : 'dataCell';
+            this.setActiveNode({ row: 0, column: 0 }, type);
             this.grid.navigateTo(0, 0, (obj) => {
                 this.grid.clearCellSelection();
                 obj.target.activate(event);
             });
         }
     }
 
-    focusFirstCell(header = true) {
+    focusFirstCell(header = true, footer = false) {
```

Comment:
why do you need to introduce this new parameter... since in the context of this method if the header param return false means that you are in the footer context. Can you think to remove the footer param?

Reply:
Because this method is used for the tBody as well, so I need to differentiate where the activeNode is exactly, otherwise when the focus is over the grid or over the trial watermark the method tracks the navigation as it is over the summary section.

Reply:
I've got what you mean. I will remove the footer parameter and I'm gonna use the header insted.